File: go/hamming/hamming.go
package hamming

import "errors"

func Distance(a, b string) (int, error) {
  differencesNum := 0

  if len(a) != len(b) {
    return differencesNum, errors.New("Sequence lengths are unequal")
  } else {
    for pos := range a {
      if a[pos] != b[pos] {
        differencesNum += 1
      }
    }

    return differencesNum, nil
  }
}

File: python/hamming/hamming.py
def distance(strand_a, strand_b):
    if len(strand_a) != len(strand_b):
        raise ValueError("Strands are different lengths")

    hamming_distance = 0

    for dna_index in range(len(strand_a)):
        if strand_a[dna_index] != strand_b[dna_index]:
            hamming_distance += 1

    return hamming_distance
